parser/wasm_parse/mod.rs

```diff
@@ -292,6 +325,176 @@ impl<'a> Parse<'a> for wasmparser::ModuleReader<'a> {
     }
 }
 
+fn get_section_name(section: &wasmparser::Section<'_>) -> String {
+    match section.code {
+        wasmparser::SectionCode::Custom { name, .. } => {
+            format!("custom section '{}' headers", name)
+        }
+        wasmparser::SectionCode::Type => "type section headers".to_string(),
+        wasmparser::SectionCode::Import => "import section headers".to_string(),
+        wasmparser::SectionCode::Function => "function section headers".to_string(),
+        wasmparser::SectionCode::Table => "table section headers".to_string(),
+        wasmparser::SectionCode::Memory => "memory section headers".to_string(),
+        wasmparser::SectionCode::Global => "global section headers".to_string(),
+        wasmparser::SectionCode::Export => "export section headers".to_string(),
+        wasmparser::SectionCode::Start => "start section headers".to_string(),
+        wasmparser::SectionCode::Element => "element section headers".to_string(),
+        wasmparser::SectionCode::Code => "code section headers".to_string(),
+        wasmparser::SectionCode::Data => "data section headers".to_string(),
+        wasmparser::SectionCode::DataCount => "data count section headers".to_string(),
+    }
+}
+
+impl<'a> Parse<'a> for (IndexedSection<'a>, IndexedSection<'a>) {
```

Comment:
Function/Code sections are merged by parsing them together. This is where most of the important changes to this file are happening.